api/api_ethereum.go

```diff
@@ -1029,3 +1284,224 @@ func (api *EthereumAPI) rpcMarshalBlock(block *types.Block, inclTx, fullTx bool)
 
 	return fields, nil
 }
+
+func EthDoCall(ctx context.Context, b Backend, args EthTransactionArgs, blockNrOrHash rpc.BlockNumberOrHash, overrides *EthStateOverride, timeout time.Duration, globalGasCap uint64) ([]byte, uint64, error) {
+	defer func(start time.Time) { logger.Debug("Executing EVM call finished", "runtime", time.Since(start)) }(time.Now())
+
+	st, header, err := b.StateAndHeaderByNumberOrHash(ctx, blockNrOrHash)
+	if st == nil || err != nil {
+		return nil, 0, err
+	}
+	if err := overrides.Apply(st); err != nil {
+		return nil, 0, err
+	}
+
+	// Setup context so it may be cancelled the call has completed
+	// or, in case of unmetered gas, setup a context with a timeout.
+	var cancel context.CancelFunc
+	if timeout > 0 {
+		ctx, cancel = context.WithTimeout(ctx, timeout)
+	} else {
+		ctx, cancel = context.WithCancel(ctx)
+	}
+	// Make sure the context is cancelled when the call has completed
+	// this makes sure resources are cleaned up.
+	defer cancel()
+
+	// TODO-Klaytn: Klaytn is using fixed baseFee as now but, if we change this fixed baseFee as dynamic baseFee, we should update this logic too.
+	fixedBaseFee := new(big.Int).SetUint64(params.BaseFee)
+	intrinsicGas, err := types.IntrinsicGas(args.data(), args.To == nil, b.ChainConfig().Rules(header.Number))
+	if err != nil {
+		return nil, 0, err
+	}
+	msg, err := args.ToMessage(globalGasCap, fixedBaseFee, intrinsicGas)
+	if err != nil {
+		return nil, 0, err
+	}
+	evm, vmError, err := b.GetEVM(ctx, msg, st, header, vm.Config{})
```

Comment:
Would you check that need to call a `state.AddBalance()` function before calling `GetEVM()`? 
You moved the code `state.AddBalance()` from `GetEVM()`.

Reply:
You can check [here](https://github.com/klaytn/klaytn/pull/1134#discussion_r802444765)